weldx/geometry.py

```diff
@@ -2374,12 +2383,31 @@ def from_geometry_raster(geometry_raster: np.ndarray) -> "SpatialData":
             return SpatialData(*ut.triangulate_geometry(geometry_raster))
 
         part_data = [ut.triangulate_geometry(part) for part in geometry_raster]
-
         total_points = []
         total_triangles = []
-        for points, triangulation in part_data:
+        for i, (points, triangulation) in enumerate(part_data):
             total_triangles += (triangulation + len(total_points)).tolist()
+            if closed_mesh:
+                # closes side faces
+                i_p1 = len(total_points)
+                i_p2 = i_p1 + len(geometry_raster[i][0][0]) - 1
+                i_p3 = i_p1 + len(points) - 1
+                i_p4 = i_p3 - len(geometry_raster[i][-1][0]) + 1
+                total_triangles += [[i_p1, i_p2, i_p3], [i_p3, i_p4, i_p1]]
+
+                # closes front and back faces
+                def _triangulate_profile(p_0, p_1):
+                    triangles = []
+                    while p_1 > p_0:
+                        triangles += [[p_0, p_1, p_1 - 1], [p_1 - 1, p_0 + 1, p_0]]
+                        p_0 += 1
+                        p_1 -= 1
+                    return triangles
```

Comment:
This is a very simple (and unelegant) algorithm that easily breaks for more complex (concave) shapes. We should replace it in the future with a prober 2d polygon triangulation algorithm. However, for our current simple profiles, it should be enough